python/ray/actor.py

```diff
@@ -140,6 +140,79 @@ def __setstate__(self, state):
             hardref=True)
 
 
+class ActorClassMethodMetadata(object):
+    """Metadata for all methods in an actor class. This data can be cached.
+
+    Attributes:
+        methods: The actor methods.
+        decorators: Optional decorators that should be applied to the
+            method invocation function before invoking the actor methods. These
+            can be set by attaching the attribute
+            "__ray_invocation_decorator__" to the actor method.
+        signatures: The signatures of the methods.
+        num_return_vals: The default number of return values for
+            each actor method.
+    """
+
+    def __init__(self):
+        class_name = type(self).__name__
+        raise Exception("{} can not be constructed directly, "
+                        "instead of running '{}()', try '{}.create()'".format(
+                            class_name, class_name, class_name))
+
+    @classmethod
+    def create(cls, modified_class, actor_creation_function_descriptor):
+        worker = ray.worker.get_global_worker()
+        # Try to create an instance from cache.
+        cached_meta = worker.cached_actor_method_metadata.get(
```

Comment:
Just put the cache dict in `cls`?

Reply:
We can clear the ray.worker's cache easily in `ray.shutdown()`. If we put the cache in cls, `worker.py` have to import `actor.py` to clear the cache.